src/client/GF256.cpp

```diff
@@ -0,0 +1,348 @@
+    (int8_t) 0x96, (int8_t) 0xdb, (int8_t) 0xbd, (int8_t) 0xf1, (int8_t) 0xd2,
+    (int8_t) 0x13, (int8_t) 0x5c, (int8_t) 0x83, (int8_t) 0x38, (int8_t) 0x46,
+    (int8_t) 0x40, (int8_t) 0x1e, (int8_t) 0x42, (int8_t) 0xb6, (int8_t) 0xa3,
+    (int8_t) 0xc3, (int8_t) 0x48, (int8_t) 0x7e, (int8_t) 0x6e, (int8_t) 0x6b,
+    (int8_t) 0x3a, (int8_t) 0x28, (int8_t) 0x54, (int8_t) 0xfa, (int8_t) 0x85,
+    (int8_t) 0xba, (int8_t) 0x3d, (int8_t) 0xca, (int8_t) 0x5e, (int8_t) 0x9b,
+    (int8_t) 0x9f, (int8_t) 0x0a, (int8_t) 0x15, (int8_t) 0x79, (int8_t) 0x2b,
+    (int8_t) 0x4e, (int8_t) 0xd4, (int8_t) 0xe5, (int8_t) 0xac, (int8_t) 0x73,
+    (int8_t) 0xf3, (int8_t) 0xa7, (int8_t) 0x57, (int8_t) 0x07, (int8_t) 0x70,
+    (int8_t) 0xc0, (int8_t) 0xf7, (int8_t) 0x8c, (int8_t) 0x80, (int8_t) 0x63,
+    (int8_t) 0x0d, (int8_t) 0x67, (int8_t) 0x4a, (int8_t) 0xde, (int8_t) 0xed,
+    (int8_t) 0x31, (int8_t) 0xc5, (int8_t) 0xfe, (int8_t) 0x18, (int8_t) 0xe3,
+    (int8_t) 0xa5, (int8_t) 0x99, (int8_t) 0x77, (int8_t) 0x26, (int8_t) 0xb8,
+    (int8_t) 0xb4, (int8_t) 0x7c, (int8_t) 0x11, (int8_t) 0x44, (int8_t) 0x92,
+    (int8_t) 0xd9, (int8_t) 0x23, (int8_t) 0x20, (int8_t) 0x89, (int8_t) 0x2e,
+    (int8_t) 0x37, (int8_t) 0x3f, (int8_t) 0xd1, (int8_t) 0x5b, (int8_t) 0x95,
+    (int8_t) 0xbc, (int8_t) 0xcf, (int8_t) 0xcd, (int8_t) 0x90, (int8_t) 0x87,
+    (int8_t) 0x97, (int8_t) 0xb2, (int8_t) 0xdc, (int8_t) 0xfc, (int8_t) 0xbe,
+    (int8_t) 0x61, (int8_t) 0xf2, (int8_t) 0x56, (int8_t) 0xd3, (int8_t) 0xab,
+    (int8_t) 0x14, (int8_t) 0x2a, (int8_t) 0x5d, (int8_t) 0x9e, (int8_t) 0x84,
+    (int8_t) 0x3c, (int8_t) 0x39, (int8_t) 0x53, (int8_t) 0x47, (int8_t) 0x6d,
+    (int8_t) 0x41, (int8_t) 0xa2, (int8_t) 0x1f, (int8_t) 0x2d, (int8_t) 0x43,
+    (int8_t) 0xd8, (int8_t) 0xb7, (int8_t) 0x7b, (int8_t) 0xa4, (int8_t) 0x76,
+    (int8_t) 0xc4, (int8_t) 0x17, (int8_t) 0x49, (int8_t) 0xec, (int8_t) 0x7f,
+    (int8_t) 0x0c, (int8_t) 0x6f, (int8_t) 0xf6, (int8_t) 0x6c, (int8_t) 0xa1,
+    (int8_t) 0x3b, (int8_t) 0x52, (int8_t) 0x29, (int8_t) 0x9d, (int8_t) 0x55,
+    (int8_t) 0xaa, (int8_t) 0xfb, (int8_t) 0x60, (int8_t) 0x86, (int8_t) 0xb1,
+    (int8_t) 0xbb, (int8_t) 0xcc, (int8_t) 0x3e, (int8_t) 0x5a, (int8_t) 0xcb,
+    (int8_t) 0x59, (int8_t) 0x5f, (int8_t) 0xb0, (int8_t) 0x9c, (int8_t) 0xa9,
+    (int8_t) 0xa0, (int8_t) 0x51, (int8_t) 0x0b, (int8_t) 0xf5, (int8_t) 0x16,
+    (int8_t) 0xeb, (int8_t) 0x7a, (int8_t) 0x75, (int8_t) 0x2c, (int8_t) 0xd7,
+    (int8_t) 0x4f, (int8_t) 0xae, (int8_t) 0xd5, (int8_t) 0xe9, (int8_t) 0xe6,
+    (int8_t) 0xe7, (int8_t) 0xad, (int8_t) 0xe8, (int8_t) 0x74, (int8_t) 0xd6,
+    (int8_t) 0xf4, (int8_t) 0xea, (int8_t) 0xa8, (int8_t) 0x50, (int8_t) 0x58,
+    (int8_t) 0xaf
+};
+
+int8_t GF256::gfMul(int8_t a, int8_t b) {
+    if ((a == 0) || (b == 0)) {
+        return 0;
+    }
+
+    int tmp = (GF_LOG_BASE[a & 0xff] & 0xff) + (GF_LOG_BASE[b & 0xff] & 0xff);
```

Comment:
There is no need to do `GF_LOG_BASE[x] & 0xff`, since every element in GF_LOG_BASE is no larger then 255